From 8ba22b48ebff50311d7eaa8d512f9d507f0bdd0d Mon Sep 17 00:00:00 2001
From: Tom Taylor <tomtaylor@google.com>
Date: Wed, 4 Jan 2017 09:42:37 -0800
Subject: 33388925  Mismatched new vs delete in framesequence library

* The array allocations neglected to include [] so delete, instead
of delete [] would get called.

* Test
Manual
- tested sending a large gif that would invoke the GifTranscoder library
to make the gif smaller.

Bug: 33388925
Change-Id: I1e200e470d66ae615ffe9340ff9c049eaa73f63c
(cherry picked from commit 837474a376a4599d57f791966080f93ca7afaf67)
---
 jni/GifTranscoder.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/jni/GifTranscoder.cpp b/jni/GifTranscoder.cpp
index a7f5c74..6245538 100644
--- a/jni/GifTranscoder.cpp
+++ b/jni/GifTranscoder.cpp
@@ -144,10 +144,10 @@ bool GifTranscoder::resizeBoxFilter(GifFileType* gifIn, GifFileType* gifOut) {
     std::vector<GifByteType> srcBuffer(gifIn->SWidth * gifIn->SHeight);
 
     // Buffer for rendering images from the input GIF.
-    std::unique_ptr<ColorARGB> renderBuffer(new ColorARGB[gifIn->SWidth * gifIn->SHeight]);
+    std::unique_ptr<ColorARGB[]> renderBuffer(new ColorARGB[gifIn->SWidth * gifIn->SHeight]);
 
     // Buffer for writing new images to output GIF (one row at a time).
-    std::unique_ptr<GifByteType> dstRowBuffer(new GifByteType[gifOut->SWidth]);
+    std::unique_ptr<GifByteType[]> dstRowBuffer(new GifByteType[gifOut->SWidth]);
 
     // Many GIFs use DISPOSE_DO_NOT to make images draw on top of previous images. They can also
     // use DISPOSE_BACKGROUND to clear the last image region before drawing the next one. We need
-- 
cgit v1.1

